internal/k8s/controllers/node_controller.go

```diff
@@ -65,16 +68,23 @@ func (r *NodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 }
 
 func (r *NodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	p := predicate.NewPredicateFuncs(
-		func(obj client.Object) bool {
-			node, ok := obj.(*v1.Node)
+	p := predicate.Funcs{
```

Comment:
nit: can you expand the commit body with the reason why we want this change?

Reply:
I squashed all the commits into a single one with a body provinding some context for this change.

Reply:
I squashed all the commits into a single one with a body provinding some context for this change.